File: Aula_04/c++/03.cpp
#include <iostream>
#include <string>

class Conta_Bancaria{
    public:
    std::string nome = "";
    int n_conta = 0;
    double saldo = 0.0;

    void m_saldo(){
        std::cout << "Saldo atual: " << saldo << std::endl;
    }

    void deposito (double valor){
        saldo = saldo + valor;
        std::cout << "Deposito realizado com sucesso" << std::endl;
    }

    void saque (double valor){
        if (valor > saldo){
            std::cout << "Saldo insuficiente" << std::endl;
        } else {
            saldo = saldo - valor;
            std::cout << "Saque realizado com sucesso" << std::endl;
        }
    }
};

int main(){
    Conta_Bancaria cb;

    std::cout << "Digite seu nome: ";
    std::cin >> cb.nome;

    std::cout << "Digite o numero da sua conta: ";
    std::cin >> cb.n_conta;

    double valor;
    std::cout << "Digite quanto dinheiro voce quer depositar: ";
    std::cin >> valor;
    cb.deposito(valor);
    cb.m_saldo();

    std::cout << "Digite quanto dinheiro voce quer sacar: ";
    std::cin >> valor;
    cb.saque(valor);

    return 0;
}

File: Aula_04/python/03.py
class Conta_Bancaria:
    def __init__ (self):
        self.nome = ""
        self.n_conta = 0
        self.saldo = 0

    def m_saldo (self):
        print(f"Saldo atual: {self.saldo}")

    def deposito (self, valor):
        self.saldo = self.saldo + valor
        print("Deposito realizado com sucesso")

    def saque (self, valor):
        if valor > self.saldo:
            print("Saldo insuficiente")
        else:
            self.saldo = self.saldo - valor
            print("Saque realizado com sucesso")
            
cb = Conta_Bancaria()
cb.nome = input("Digite seu nome: ")
cb.n_conta = int(input("Digite o numero da sua conta: "))
cb.m_saldo()

valor = int(input("Digite quanto dinheiro voce quer depositar: "))
cb.deposito(valor)
cb.m_saldo()

valor = int(input("Digite quanto dinheiro voce quer sacar: "))
cb.saque(valor)